modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -503,6 +505,38 @@ public String toOperationId(String operationId) {
         return StringUtils.underscore(sanitizedOperationId);
     }
 
+    @Override
+    public CodegenOperation fromOperation(String path, String httpMethod, Operation operation, List<Server> servers) {
+        CodegenOperation op = super.fromOperation(path, httpMethod, operation, servers);
+
+        // If this endpoint has a single binary body parameter, we want
+        // to accept a body argument with the trait bound Into<Body>.
+        // This requires a type parameter in the function definition,
+        // in advance of actual parameter, and is thus far easier to decide
+        // here than in the template.
+        if (REQWEST_LIBRARY.equals(getLibrary()) && op.bodyParams != null && op.bodyParams.size() == 1) {
+            CodegenParameter cp = op.bodyParams.get(0);
+
+            if (cp.isBinary) {
+                cp.baseType = cp.dataType = "B";
+                op.vendorExtensions.put("x-rust-type-parameter", "B: Into<reqwest::Body>");
+
+                // A copy of the body parameter will appear somewhere in the
+                // allParams list.  We need to doctor that one as well.
+                int seen = 0;
+                for (CodegenParameter ap : op.allParams) {
+                    if (ap.isBinary && ap.isBodyParam) {
+                        ap.baseType = ap.dataType = "B";
+                        seen++;
+                    }
+                }
+                assert seen == 1;
```

Comment:
Should we use `assert` in test code only?

Reply:
What about simply using `isBinary` in the mustache template instead of creating `x-rust-type-parameter` ?

Reply:
> Should we use `assert` in test code only?

I put the assert here because it should be an invariant.  The generated code won't be correct (something is seriously wrong) if there is more than one binary body parameter in the `allParams` list, when there was only one in the `bodyParams` list.  It seems like it would be better to crash rather than to generate incorrect code if that is ever true, as it presumably represents a bug in the code.

> What about simply using `isBinary` in the mustache template instead of creating `x-rust-type-parameter` ?

I'm not sure how to do that with the limited functionality available in the templates.  We need three things to be true:

- that it's binary
- that it's a body parameter
- that it's the only body parameter

In addition to these three conditions, we need to make that determination _outside_ of the `{{#allParams}}` loop in the template where we emit function arguments.  I could not see how to use the `isBinary` property of a specific index in the `bodyParams` list as a predicate in the appropriate part of the template, or how to use the length of the list as a predicate.

I also expect that as the Rust code generator becomes more complete we will need to do more of this.  That is, add more trait bounds and generic parameters as certain other types of arguments are added; e.g., today for strings we generate code like:

```rust
async fn endpoint(
    arg1: &str,
    arg2: &str,
)
```

But it would be more common to accept something like an `AsRef<str>`; e.g.,

```rust
async fn endpoint<S1: AsRef<str>, S2: AsRef<str>>(
    arg1: S1,
    arg2: S2,
)
```

This means you can pass either a `&str` or a `String` or, I believe, something that implements `Display`.

What are your thoughts?

Reply:
>I put the assert here because it should be an invariant. The generated code won't be correct (something is seriously wrong)

Don't think we should use assert in non-test code. We usually use condition to perform a check and throw an exception instead.

Reply:
@wing328 which assertion should be raised here?

Reply:
maybe just a RuntimeException